alacritty_terminal/src/tty/windows/child.rs

```diff
@@ -32,7 +33,13 @@ extern "system" fn child_exit_callback(ctx: *mut c_void, timed_out: BOOLEAN) {
     }
 
     let event_tx: Box<_> = unsafe { Box::from_raw(ctx as *mut ChildExitSender) };
-    let _ = event_tx.sender.send(ChildEvent::Exited);
+
+    let mut exit_code = 0_u32;
+    let child_handle = event_tx.child_handle.load(Ordering::Relaxed) as HANDLE;
+    let status = unsafe { GetExitCodeProcess(child_handle, &mut exit_code) };
+    let exit_code = if status > 0 { Some(exit_code as i32) } else { None };
```

Comment:
> If the function succeeds, the return value is nonzero.

The return type of `GetExitCodeProcess` is an integer and the documentation states that anything nonzero indicates success. As such it would be possible to get a return value of `-1`, which is smaller than 0 but still a valid success return value.

Generally you want to handle stuff like this using `status != FALSE`, where `FALSE` is the constant imported from windows_sys.

Reply:
Indeed, thank you for catching that, fixed.